plugins/telescope/file-browser.nix

```
@@ -0,0 +1,213 @@
+{
+  pkgs,
+  config,
+  lib,
+  ...
+}:
+with lib; let
+  cfg = config.plugins.telescope.extensions.file_browser;
+  helpers = import ../helpers.nix {inherit lib;};
+
+  hiddenOption = types.submodule {
+    options = {
+      fileBrowser = helpers.mkNullOrOption types.bool "";
+
+      folderBrowser = helpers.mkNullOrOption types.bool "";
+    };
+  };
+
+  statOption = types.submodule {
+    options = {
+      width = helpers.mkNullOrOption types.int "";
+      display = helpers.mkNullOrOption types.str "";
+      hl = helpers.mkNullOrOption types.str "";
+    };
+  };
+in {
+  options.plugins.telescope.extensions.file_browser = {
+    enable = mkEnableOption "File browser extension for telescope";
+
+    package = helpers.mkPackageOption "telescope file_browser" pkgs.vimPlugins.telescope-file-browser-nvim;
+
+    theme = helpers.mkNullOrOption types.str "Custom theme, will use your global theme by default.";
+
+    path = helpers.defaultNullOpts.mkNullable (types.either types.str helpers.rawType) "vim.loop.cwd()" ''
+      dir to browse files from, `vim.fn.expanded` automatically
```

```suggestion
      Directory to browse files from.
      `vim.fn.expanded` automatically.
```

Could you start comments with a capital letter and end them with a period please ?
Sorry for bothering you with that but it is to keep the documentation coherent.

---

Sure thing, i tried to use other telescope extensions as a reference and it seems that wasn't a good choice lol